src/ripple/nodestore/impl/Shard.cpp

```diff
@@ -590,9 +591,33 @@ Shard::isLegacy() const
 }
 
 bool
-Shard::finalize(
-    bool const writeSQLite,
-    boost::optional<uint256> const& expectedHash)
+Shard::finalize(bool writeSQLite, boost::optional<uint256> const& expectedHash)
+{
+    auto const scopedCount{makeBackendCount()};
+    if (!scopedCount)
+        return false;
+
+    state_ = finalizing;
+
+    /** The finalizeInternal() function verifies the shard and, if third
+     * parameter is true, then replaces the shard by deterministic copy
+     * of the shard. After deterministic shard is created, it verifies
+     * deterministic shard again, third parameter is false.
+     */
+    if (!finalizeInternal(writeSQLite, expectedHash, true))
+        return false;
+    if (!finalizeInternal(false, expectedHash, false))
```

Comment:
Is there a specific reason we verify the deterministic shard again? I don't see this being necessary. Unless I am missing something, the third parameter can be removed and the deterministic shard should always be created.

Reply:
@cdy20 Here is a branch with the changes I mentioned above. Feel free to cherry-pick or copy etc... https://github.com/miguelportilla/rippled/commits/d_shard

Reply:
Done.